nifi-nar-bundles/nifi-camel-bundle/nifi-camel-processors/src/main/java/org/apache/nifi/processors/camel/CamelProcessor.java

```diff
@@ -0,0 +1,240 @@
+import org.springframework.context.support.ClassPathXmlApplicationContext;
+import org.springframework.context.support.FileSystemXmlApplicationContext;
+import org.springframework.context.support.GenericXmlApplicationContext;
+import org.springframework.core.io.ByteArrayResource;
+
+/**
+ * This processor runs a Camel Route.
+ */
+@Tags({"camel", "route", "put"})
+@InputRequirement(Requirement.INPUT_ALLOWED)
+@CapabilityDescription("Runs a Camel Route. Each input FlowFile is converted into a Camel Exchange "
+                       + "for processing by configured Route. It exports ProcessSession to camel exchange header 'nifiSession'")
+public class CamelProcessor extends AbstractProcessor {
+
+    protected static final Relationship SUCCESS = new Relationship.Builder().name("success")
+        .description("Camel Route has Executed Successfully").build();
+
+    protected static final Relationship FAILURE = new Relationship.Builder().name("failure")
+        .description("Camel Route has Failed to Execute").build();
+
+    public static final PropertyDescriptor CAMEL_SPRING_CONTEXT_FILE_PATH = new PropertyDescriptor.Builder()
+        .name("Camel Spring Config File Path")
+        .description("The Classpath where NiFi can find Spring Application context file"
+                         + " Ex: classpath:/META-INF/camel-application-context.xml")
+        .defaultValue("classpath:/META-INF/camel-application-context.xml").required(true).addValidator(Validator.VALID)
+        .build();
+
+    public static final PropertyDescriptor CAMEL_SPRING_CONTEXT_DEF = new PropertyDescriptor.Builder()
+    .name("Camel Spring Context Definition")
+    .description("Content of Spring Application context ")
+    .defaultValue("").addValidator(Validator.VALID)
+    .build();
+
+    public static final PropertyDescriptor CAMEL_ENTRY_POINT_URI = new PropertyDescriptor.Builder()
+        .name("Camel EntryPoint")
+        .description("EntryPoint for NiFi in Camel Route" + " Ex: direct-vm:nifiEntryPoint")
+        .defaultValue("direct-vm:nifiEntryPoint").required(true).addValidator(Validator.VALID).build();
+
+    /**
+     *@see <a href="http://camel.apache.org/grape.html">  Camel Grape Documentation</a>
+     */
+    public static final PropertyDescriptor EXT_LIBRARIES = new PropertyDescriptor.Builder()
+    .name("Extra Libraries")
+    .description("Comma Seperated List of Extra Libraries/Features to Download and Use [ in GroupId/ArtifactId/version format]. "
+        + "Ex: org.apache.camel/camel-mail/2.16.1,\n"
+        + "org.apache.camel/camel-infinispan/2.16.1")
+    .required(false).addValidator(new GrapeGrabValidator()).build();
+
+    private static SpringCamelContext camelContext = null;
+
+    private ImmutableList<PropertyDescriptor> descriptors;
+
+    private ImmutableSet<Relationship> relationships=ImmutableSet.of(SUCCESS, FAILURE);
+
+    public static synchronized SpringCamelContext getCamelContext() {
```

Comment:
I am not sure I understand why is this a public method. Are there expectations of accessing context from outside? I mean sure if at some point someone would extend from this class it may be useful and we can alway relax the visibility, but at this point I don't see the reason. 
Also, why is it static and why is it synchronized?